### PR TITLE
[GLUTEN-2090] fix type id cast bug

### PR DESCRIPTION
**Pull request by binmahone**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by binmahone** [comment]
Run Gluten Clickhouse CI

**Comment by lgbo-ustc** [comment]
LGTM

**Review by baibaichen — APPROVED** [review]
LGTM

File: cpp-ch/local-engine/Parser/SerializedPlanParser.cpp [hunk 1]
```diff
@@ -442,7 +442,7 @@ Block SerializedPlanParser::parseNameStruct(const substrait::NamedStruct & struc
             auto tmp_ctx = DB::Context::createCopy(global_context);
             SerializedPlanParser tmp_plan_parser(tmp_ctx);
             auto function_parser = FunctionParserFactory::instance().get(name_parts[3], &tmp_plan_parser);
-            auto agg_function_parser = typeid_cast<BaseAggregateFunctionParser *>(function_parser.get());
+            auto agg_function_parser = dynamic_cast<BaseAggregateFunctionParser *>(function_parser.get());
```
Comment (lgbo-ustc) [inline cpp-ch/local-engine/Parser/SerializedPlanParser.cpp:445]: 这里可以去掉

**Pull request merged**